app/controllers/application_controller.rb

```diff
@@ -252,6 +252,32 @@ def initiate_wait_for_task(options = {})
   end
   private :initiate_wait_for_task
 
+  def generate_gtl
```

Comment:
I would suggest extracting lines 257-266 as a separate method such as
```
def view_model_from_params(active_tree, model, controller)
```
so that the (quite comple) logic of finding the view would be separated from the part that prepares and renders the json data.

In short: this method does too many things and it will be hard to test and maintain.